### PR TITLE
init: improve handling of import errors

### PR DESCRIPTION
**Pull request by blueyed**
See the individual commits.

TODO:
- [ ] squash before merging

### CODE REVIEW

File: pythonx/jedi_vim.py [hunk 1]
```diff
@@ -110,10 +110,14 @@ def echo_highlight(msg):
...
-except ImportError as e:
-    no_jedi_warning(str(e))
+except ImportError:
     jedi = None
-    jedi_import_error = str(e)
+    _, exc, tb = sys.exc_info()
+    while tb.tb_next:
+        tb = tb.tb_next
+    f_code = tb.tb_frame.f_code
+    exc_location = '%s:%d' % (f_code.co_filename, f_code.co_firstlineno)
```
Comment (davidhalter) [inline pythonx/jedi_vim.py]: Why don't we just return the whole traceback with `import traceback`? Wouldn't that be nicer and less code?

Comment (blueyed) [reply pythonx/jedi_vim.py]: Good point, but not trivial.  I will improve this.

**Comment by blueyed** [comment]
Also uses PythonToVimStr now for saner string handling.

File: pythonx/jedi_vim_debug.py [hunk 2]
```diff
@@ -1,57 +1,78 @@
...
+    echo(' - global site module: `{0}`'.format(__import__('site').__file__))
 
     try:
         import jedi_vim
-    except Exception as e:
-        echo("printf('ERROR: jedi_vim is not available: %s: %s', "
-             "{0!r}, {1!r})".format(e.__class__.__name__, str(e)))
+    except Exception:
+        echo_error('ERROR: jedi_vim is not available: {0}'.format(
+            format_exc_info(sys.exc_info())))
```
Comment (davidhalter) [inline pythonx/jedi_vim_debug.py]: We can probably just remove sys.exc_info here, right?

Comment (blueyed) [reply pythonx/jedi_vim_debug.py]: Yeah, I've added the default handling afterwards apparently.

Comment (blueyed) [reply pythonx/jedi_vim_debug.py]: Fixed.

**Pull request merged**